src/wallet/rpcwallet.cpp

```diff
@@ -2615,7 +2615,18 @@ static RPCHelpMan loadwallet()
     if (!wallet) {
         // Map bad format to not found, since bad format is returned when the
         // wallet directory exists, but doesn't contain a data file.
-        RPCErrorCode code = status == DatabaseStatus::FAILED_NOT_FOUND || status == DatabaseStatus::FAILED_BAD_FORMAT ? RPC_WALLET_NOT_FOUND : RPC_WALLET_ERROR;
+        RPCErrorCode code = RPC_WALLET_ERROR;
+        switch (status) {
+            case DatabaseStatus::FAILED_NOT_FOUND:
+            case DatabaseStatus::FAILED_BAD_FORMAT:
+                code = RPC_WALLET_NOT_FOUND;
+                break;
+            case DatabaseStatus::FAILED_ALREADY_LOADED:
+                code = RPC_WALLET_ALREADY_LOADED;
+                break;
+            default:
```

Comment:
Maybe set `code` to `RPC_WALLET_ERROR` in default instead of initiliasing it?
```suggestion
            default:
                code = RPC_WALLET_ERROR;
```

Reply:
What would be the difference? I personally prefer variables to be initialized where they are defined, so that if code later change the risk of introducing undefined behavior is smaller.

Reply:
No difference, just makes it more clear that by default the error code is RPC_WALLET_ERROR.

Reply:
Sure. I'll add a comment.